File: Solution/WebSolution/AssessmentPortal/Monolithic/Monolithic_Controller/AssessmentEntities/SubjectMatterExpert.cs
using System;

namespace Transflower.TFLAssessment.Entities
{
    public class SubjectMatterExpert
    {
        public int Id { get; set; }
        public int SubjectId { get; set; }
        public int EmployeeId { get; set; }

        // Navigation Properties
        public Subject Subject { get; set; }
        public Employee Employee { get; set; }
    }
}


package com.transflower.tflAssesment.entities;

public class SubjectMatterExpert {

    private int Id;
    private int SubjectId;
    private int EmployeeId;
    private Subject Subject;
    private Employee Employee;

    public int getId(int Id)
    {
        return Id;
    }

    public void setId()
    {
        this.Id = Id;
    }

    public int getSubjectId(int SubjectId)
    {
        return SubjectId;
    }
    public void setSubjectId()
    {
        this.SubjectId = SubjectId;
    }
    public int getEmployeeId(int EmployeeId)
    {
        return EmployeeId;
    }
    public int setEmployeeId()
    {
        this.EmployeeId = EmployeeId;
    }
    public Subject getSubject(String Subject)
    {
        return Subject;
    }
    public Subject setSubject()
    {
        this.Subject = Subject;
    }
    public Employee getEmployee(Employee Employee)
    {
        return Employee;
    }
    public Employee setEmployee()
    {
        this.Employee = Employee;
    }
}

File: Solution/others/Java/tflassessment-demo/src/main/java/com/transflower/tflassessment/demo/entities/SubjectMatterExpert.java
package com.transflower.tflassessment.demo.entities;

import java.util.Objects;

import javax.security.auth.Subject;

public class SubjectMatterExpert implements Cloneable {

    private int id;
    private int subjectId;
    private int employeeId;
    private Subject subject;
    private Employee employee;

    public SubjectMatterExpert(int id, int subjectId, int employeeId, Subject subject, Employee employee) {
        this.id = id;
        this.subjectId = subjectId;
        this.employeeId = employeeId;
        this.subject = subject;
        this.employee = employee;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getSubjectId() {
        return subjectId;
    }

    public void setSubjectId(int subjectId) {
        this.subjectId = subjectId;
    }

    public int getEmployeeId() {
        return employeeId;
    }

    public void setEmployeeId(int employeeId) {
        this.employeeId = employeeId;
    }

    public Subject getSubject() {
        return subject;
    }

    public void setSubject(Subject subject) {
        this.subject = subject;
    }

    public Employee getEmployee() {
        return employee;
    }

    public void setEmployee(Employee employee) {
        this.employee = employee;
    }

    @Override
    public String toString() {
        return "SubjectMatterExpert{" +
                "id=" + id +
                ", subjectId=" + subjectId +
                ", employeeId=" + employeeId +
                ", subject=" + subject +
                ", employee=" + employee +
                '}';
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;
        SubjectMatterExpert other = (SubjectMatterExpert) obj;
        return id == other.id &&
                subjectId == other.subjectId &&
                employeeId == other.employeeId &&
                Objects.equals(subject, other.subject) &&
                Objects.equals(employee, other.employee);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, subjectId, employeeId, subject, employee);
    }

    @Override
    protected void finalize() throws Throwable {
        try {
            System.out.println("Finalize method called: " + this);
        } finally {
            
        }
    }

    @Override
    public Object clone() throws CloneNotSupportedException {
        return super.clone();
    }

}
